omniledger/service/transaction.go

```diff
@@ -353,6 +366,27 @@ func (sc StateAction) String() string {
 	}
 }
 
+type instrType int
+
+const (
+	spawnType instrType = iota
+	invokeType
+	deleteType
+	invalidInstrType
```

Comment:
Normally I'd expect the zero value of a type to be the invalid one, so that in case the zero value is accidentally used someplace, it does not accidentally mean "spawn".

This new getType is really nice, thanks for doing it private only, very good API cleanliness. :)